### PR TITLE
README: Create base skeleton of language specification

### PR DESCRIPTION
**Pull request by Hamzantal**
This is just a base skeleton, the items need to be reordered.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1 +1,61 @@
...
+
+- Focus for support for embedded usage?
+
+- No interfaces
+
+- Python like modules?
+
+- Sled -> package manager
+
+- Sled uses centralized package system?
```
Comment (Henrixounez) [inline readme.md]: Sled might be only a proxy to github pages

File: readme.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+- regular bitwise operators:
+ 
+```
+BITWISE
+&        and
+|        or
+^        xor
+>>       right shift
+<<       left shift
+!        no
```
Comment (Henrixounez) [inline readme.md]: ~  is no bitwise operator

**Comment by LouCle** [comment]
also please use these datatypes
```
B       Type
-       -
1       byte
1       char
1       bool
2       short
4       int
4       float
8       long
8       double
12      somename (long double)
```

**Comment by lutherwenxu** [comment]
Reword your commit to fit in the `module: what you did` requirement.
`README: add discussed specs into readme` would fit nicely for this PR.

**Comment by superwhiskers** [comment]
@LouCle we're using i<bit-length> as the integer type and u<bit-length> as the unsigned integer type

**Comment by Hamzantal** [comment]
I added changes. Please confirm @Henrixounez @superwhiskers @lutherwenxu @LouCle

File: readme.md [hunk 3]
```diff
@@ -1 +1,77 @@
...
+BITWISE
+&        and
+|        or
+^        xor
+>>       right shift
+<<       left shift
+~        no
+```
+
+- For equations with multiple types, bigger type wins
```
Comment (lutherwenxu) [inline readme.md]: This is not true, we require manual type casting. No more inferred bigger type wins.

Comment (LouCle) [reply readme.md]: Exactly.

```c
foo i8 := 2 + 2.5 // --> 4, as the float 2.5 is also typecasted to i8
bar f32 := 2 + 2.5 // --> 4.5
```
I don't know what we should do in the case of
```c
foo := 2 + 2.5
```
Perhaps we should avoid type inference all together

Comment (lutherwenxu) [reply readme.md]: That would be a compilation error. I mean maybe they want it converted to float, do some math and then cast it back to i8. No one knows.

Comment (LouCle) [reply readme.md]: I agree
```c
foo := 2 + 2.5 // --> error
```

**Pull request merged**